Assets/BossRoom/Scripts/Shared/Net/NetworkStats.cs

```diff
@@ -16,6 +16,35 @@ namespace Unity.Multiplayer.Samples.BossRoom
     [RequireComponent(typeof(NetworkObject))]
     public class NetworkStats : NetworkBehaviour
     {
+        private class MovingWindowAverage
```

Comment:
For a value like rtt I'd suggest to use an exponential moving average as it's a better indication of the current rtt and fluctuates less. So something like:
```cs
    public struct FloatEmaCalculator
    {
        private readonly float m_Alpha;
        private float m_Average;

        public float Average => m_Average;

        public FloatEmaCalculator(int lookBack, float average)
        {
            m_Alpha = 2f / (lookBack + 1);
            m_Average = average;
        }

        public float NextValue(float value) => m_Average = (value - m_Average) * m_Alpha + m_Average;
    }
```

Reply:
agreed, done